lrlex/src/lib/parser.rs

```diff
@@ -332,7 +333,39 @@ where
         errs: &mut Vec<LexBuildError>,
     ) -> LexInternalBuildResult<usize> {
         loop {
-            i = self.parse_ws(i)?;
+            // We should be at newline of the previous section separator '%%<here>\n upon entry,
+            // otherwise after iterating before the newline of the previous iterations rule or at eof.
+            i = self.parse_nl(i)?;
+            // According to posix lex:
+            //
+            // > Any such input (beginning with a <blank> or within "%{" and "%}" delimiter lines)
+            // > appearing at the beginning of the Rules section before any rules are specified
+            // > shall be written to lex.yy.c
+            //
+            // > The reason for the undefined condition associated with text beginning with a <blank> or within "%{" and "%}"
+            // > delimiter lines appearing in the Rules section is historical practice.
+            //
+            // > Both the BSD and System V lex copy the indented (or enclosed) input in the Rules section (except at the beginning)
+            // > to unreachable areas of the yylex() function (the code is written directly after a break statement).
+            // > In some cases, the System V lex generates an error message or a syntax error, depending on the form of indented input.
+            //
+            // Previously we allowed these, and trimmed leading spaces, parsing any rules after
```

Comment:
Can probably merge this line with the next?

Reply:
I did this and also evened out the line lengths on one of the paragraphs above it.